applications/nrf_desktop/doc/dvfs.rst

```diff
@@ -0,0 +1,104 @@
+.. _nrf_desktop_dvfs:
+
+DVFS module
+###########
+
+.. contents::
+   :local:
+   :depth: 2
+
+The nRF Desktop Dynamic Voltage and Frequency Scaling (DVFS) module is responsible for switching the frequency and voltage according to the application needs.
+DVFS is critical for balancing performance and power consumption, especially in battery-operated devices.
+
+In the nRF Desktop, the DVFS module enables the system to use the lowest possible frequency and voltage when the application performs less computation-challenging activities such as Bluetooth LE advertising.
+However, if performing more computation-consuming tasks nRF Desktop DVFS module will dynamically change to a higher frequency to increase performance.
+
+Requesting for frequency change is done by requesting clock from clock control module.
+Clock control module is responsible for choosing the right clock frequency and voltage.
+The DVFS is set by the sysctrl core and Clock control module requests are send through nrfs service DVFS.
+
+.. note::
+   In the current solution, it is assumed that no other module requests frequency changes.
+   The nRF Desktop DVFS module is the sole entity responsible for managing frequency and voltage adjustments.
+
+Supported devices
+*****************
+
+The module is only available for nrf54h20 SoC.
+Each device has a different set of available operating points.
+
+For example the nrf54h20 SoC has the following available operating points (in documentation called frequency):
+
+1. DVFS_FREQ_HIGH - high frequency - 320 MHz, 0.8 V
+#. DVFS_FREQ_MEDLOW - medium-low frequency - 128 MHz, 0.6V
+#. DVFS_FREQ_LOW - low frequency - 64 MHz, 0.5V
+
+Module events
+*************
+
+.. include:: event_propagation.rst
+    :start-after: table_dvfs_start
+    :end-before: table_dvfs_end
+
+.. note::
+    |nrf_desktop_module_event_note|
+
+Configuration
+*************
+
+The nRF Desktop Dynamic Voltage and Frequency Scaling (DVFS) module is enabled by default on supported board targets and can be disabled by setting the :option:`CONFIG_DESKTOP_DVFS` Kconfig option to ``n``.
+
+The module can be configured using the following Kconfig options:
+
+The nRF Desktop DVFS module defines independent states that have different requirements for the DVFS operating point.
```

Comment:
```suggestion
The nRF Desktop DVFS module defines independent application states that require higher clock frequency.
```

Reply:
My version seams better IMO.

Reply:
Can we add just the `application states` part then?

Reply:
I tried to separate the module from voltage control (we started using the clock control APIs; voltage is controlled under the hood from module's perspective), but I noticed that you prefer to refer to `DVFS operating points` etc. I would not insist here.

Reply:
To be precise frequency is also controlled under the hood. We have points(freq, V), which are ordered - the way we call them doesn't matter. We could name them a,b,c,...   and let user select one. The way you try to present this shows false impression that user have possibility to choose any frequency. We call them frequency because for most people it is the most important part.

Reply:
> Can we add just the `application states` part then?

isn't it already done?

Reply:
> To be precise frequency is also controlled under the hood. We have points(freq, V), which are ordered - the way we call them doesn't matter. We could name them a,b,c,... and let user select one. The way you try to present this shows false impression that user have possibility to choose any frequency. We call them frequency because for most people it is the most important part.

I am looking from clock control API user perspective (application module is clock control API user). From this perspective we control the clock frequency. Keep in mind that even back in time when we used enum, the enum names were related to frequency (`DVFS_FREQ_xyz`). So we control frequency here and voltage is scaled under the hood (application module never explicitly selected an oppoint).